codebase/src/java/disco/org/openlvc/disco/application/deadreckoning/DeadReckoningModel.java

```diff
@@ -0,0 +1,147 @@
+	//----------------------------------------------------------
+
+	//----------------------------------------------------------
+	//                   INSTANCE VARIABLES
+	//----------------------------------------------------------
+	protected DrmState initialState;
+	private double initialTime; // s
+
+	//----------------------------------------------------------
+	//                      CONSTRUCTORS
+	//----------------------------------------------------------
+	protected DeadReckoningModel( IDeadReckoningEnabled object )
+	{
+		this.initialState = new DrmState( object );
+		this.initialTime = object.getLocalTimestamp() / 1000.0;
+	}
+
+	//----------------------------------------------------------
+	//                    INSTANCE METHODS
+	//----------------------------------------------------------
+	public abstract DeadReckoningAlgorithm getAlgorithm();
+
+	public DrmState getInitialState()
+	{
+		return new DrmState( this.initialState );
+	}
+
+	/**
+	 * Returns the stae of the dead-reckoning model when extrapolated for the given duration.
+	 * 
+	 * @param dt time passed, in s
+	 * @return
+	 */
+	public abstract DrmState getStateAfter( double dt );
+
+	/**
+	 * Returns the state of the dead-reckoning model when extrapolated to the given local
+	 * timestamp.
+	 * 
+	 * @param localTimestamp target timestamp, in ms since epoch
+	 * @return
+	 */
+	public DrmState getStateAtLocalTime( long localTimestamp )
+	{
+		double targetTime = localTimestamp / 1000.0;
+		double dt = targetTime - this.initialTime;
+		return getStateAfter( dt );
+	}
+
+	// q_DR in the spec
+	protected Quaternion makeRotationQuaternion( double dt )
+	{
+		Vec3 rotAxis = new Vec3( this.initialState.angularVelocity ).normalize();
+
+		double halfBeta = (this.initialState.angularVelocity.length() * dt) / 2.0;
+		double sinHalfBeta = Math.sin( halfBeta );
+
+		double w = Math.cos( halfBeta );
+		double x = rotAxis.x * sinHalfBeta;
+		double y = rotAxis.y * sinHalfBeta;
+		double z = rotAxis.z * sinHalfBeta;
+
+		return new Quaternion( w, x, y, z );
+	}
+
+	////////////////////////////////////////////////////////////////////////////////////////////
+	/////////////////////////////// Accessor and Mutator Methods ///////////////////////////////
+	////////////////////////////////////////////////////////////////////////////////////////////
+	public long getInitalLocalTimestamp()
+	{
+		return (long)(this.initialTime * 1000.0);
+	}
+
+	//----------------------------------------------------------
+	//                     STATIC METHODS
+	//----------------------------------------------------------
+	/**
+	 * @param object
+	 * @param logger a logger to output warnings to, or `null`
+	 * @return
+	 */
+	public static DeadReckoningModel makeDeadReckoningModel( IDeadReckoningEnabled object, Logger logger )
+	{
+		DeadReckoningParameter params = object.getDeadReckoningParams();
+		switch( params.getDeadReckoningAlgorithm() )
+		{
+			case Other:
```

Comment:
I think it reads better if the `default` case block is at the end. You can read it as "if all else fails, this is what happens". I'd keep the Other and Static cases with it as well.